src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```
@@ -491,13 +538,28 @@
     <Message Importance="normal" Text="Sonar: ($(MSBuildProjectFile)) Analysis configured successfully with $(SonarProjectConfigFilePath)." />
   </Target>
 
-  <Target Name="SetRazorCodeAnalysisProperties" BeforeTargets="RazorCoreCompile" AfterTargets="CopyFilesToOutputDirectory">
+  <!-- **************************************************************************** -->
+  <!-- This target is only run in case a Razor compilation is happening.
+       This target sets the Razor specific error log properties and copies the results of the core compilation.
+       The *.pb files will be copied to ProjectSpecificOutDir.tmp directory as during the RazorCoreCompilation the *.pb
+       files will be replaced by the ones for Razor.
+
+       We want to run after the core "RazorCoreCompile" to copy the content before it is overwriten.
+  -->
+  <!-- **************************************************************************** -->
+  <Target Name="SetRazorCodeAnalysisProperties" BeforeTargets="RazorCoreCompile" AfterTargets="CopyFilesToOutputDirectory" Condition="$(RazorCompileOnBuild) == 'true' " >
     <PropertyGroup Condition=" $(ErrorLog) == $(SonarCompileErrorLog) ">
       <!-- Keep the path so that we could set SonarQubeSetting with sonar.language.roslyn.reportFilePaths -->
       <RazorSonarCompileErrorLog>$(TargetDir)$(TargetName)$(RazorTargetNameSuffix)$(TargetExt).RoslynCA.json</RazorSonarCompileErrorLog>
       <ErrorLog>$(RazorSonarCompileErrorLog)</ErrorLog>
       <RazorCompilationErrorLog>$(RazorSonarCompileErrorLog)</RazorCompilationErrorLog>
     </PropertyGroup>
+    <PropertyGroup>
+      <TemporaryProjectSpecificOutDir>$(ProjectSpecificOutDir).tmp</TemporaryProjectSpecificOutDir>
```

It's a new property, so let's use `Sonar` prefix to know that it's ours.
```suggestion
      <SonarTemporaryProjectSpecificOutDir>$(ProjectSpecificOutDir).tmp</SonarTemporaryProjectSpecificOutDir>
```